ui/inspectors/service.reel/ups-service.reel/ups-service.js

```diff
@@ -1,7 +1,47 @@
-var Component = require("montage/ui/component").Component;
-
+var Component = require("montage/ui/component").Component,
+    FreeNASService = require("core/service/freenas-service").FreeNASService,
+    Model = require("core/model/model").Model,
+    ServiceUpsShutdownmode = require("core/model/enumerations/service-ups-shutdownmode").ServiceUpsShutdownmode,
+    ServiceUpsMode = require("core/model/enumerations/service-ups-mode").ServiceUpsMode;
 /**
  * @class UpsService
  * @extends Component
  */
-exports.UpsService = Component.specialize();
+exports.UpsService = Component.specialize({
+
+    enterDocument: {
+        value: function (isFirstTime) {
+            var self = this,
+                loadingPromises = [];
+            if(isFirstTime) {
+                this.isLoading = true;
+                this.driverOptions = [];
+                this.driverPortOptions = [];
+                this.modeOptions = ServiceUpsMode.members.map(function (x) {
+                    return {
+                        label: x,
+                        value: x
+                    };
+                });
+                this.shutdownModeOptions = ServiceUpsShutdownmode.members.map(function (x) {
+                    return {
+                        label: x,
+                        value: x
+                    };
+                })
+                loadingPromises.push(
```

Comment:
Same here + return a promise that resolves when those two promises have resolved.